src/charm.py

```diff
@@ -2316,8 +2320,30 @@ def relations_user_databases_map(self) -> dict:
                     user, current_host=self.is_connectivity_enabled
                 )
             )
+
+        # Copy relations users directly instead of waiting for them to be created
+        for relation in self.model.relations[self.postgresql_client_relation.relation_name]:
+            user = f"relation_id_{relation.id}"
+            if user not in user_database_map and (
+                database := self.postgresql_client_relation.database_provides.fetch_relation_field(
+                    relation.id, "database"
+                )
+            ):
+                user_database_map[user] = database
         return user_database_map
 
+    @property
+    def generate_user_hash(self) -> str:
```

Comment:
Tried to hash the `relations_user_databases_map` property, but that never idled in the new relation integration test.